File: backend/retry.js
const fs = require('fs');

module.exports = (req, res) => {
  const queue = JSON.parse(fs.readFileSync('fallback/fallback-queue.json'));
  const newQueue = [];

  queue.forEach(entry => {
    const file = `targets/${entry.target}`;
    const marker = `@insert:${entry.tag}`;
    try {
      let content = fs.readFileSync(file, 'utf8');
      if (!content.includes(marker)) throw 'Marker not found';
      content = content.replace(marker, `${marker}\n${entry.snippet}`);
      fs.writeFileSync(file, content);
      fs.appendFileSync('changelog/insertion-log.txt', `${new Date()} | ${entry.tag} → ${entry.target}\n`);
    } catch (e) {
      newQueue.push(entry);
    }
  });

  fs.writeFileSync('fallback/fallback-queue.json', JSON.stringify(newQueue, null, 2));
  res.send('✅ Retry complete');
};

File: src/retry.py
import json
from utils import insert_snippet, log_injection

with open("fallback/fallback-queue.json") as f:
    queue = json.load(f)

new_queue = []
for entry in queue:
    try:
        insert_snippet(f"targets/{entry['target']}", entry["tag"], entry["snippet"])
        log_injection(entry["tag"], entry["target"])
        print(f"✅ Retried: {entry['tag']} → {entry['target']}")
    except Exception as e:
        print(f"❌ Retry failed: {entry['tag']} → {e}")
        new_queue.append(entry)

with open("fallback/fallback-queue.json", "w") as f:
    json.dump(new_queue, f, indent=2)